tools/project.py

```diff
@@ -790,15 +836,17 @@ def write_custom_step(step: str, prev_step: Optional[str] = None) -> None:
                     dyndep=custom_step.get("dyndep", None),
                 )
                 n.newline()
+
         n.build(
             outputs=step,
             rule="phony",
             inputs=implicit,
             order_only=prev_step,
+            implicit=extra_inputs,
         )
 
     # Add all build steps needed before we compile (e.g. processing assets)
-    write_custom_step("pre-compile")
+    write_custom_step("pre-compile", extra_inputs=[get_pch_out_name(config, pch) for pch in config.precompiled_headers])
```

Comment:
This line errors out when `precompiled_headers` is the default `None` value, this fixes it:
```suggestion
    write_custom_step("pre-compile", extra_inputs=[get_pch_out_name(config, pch) for pch in config.precompiled_headers or []])
```

Reply:
Also let's split this into a separate line